pyannote/metrics/base.py

```diff
@@ -247,6 +247,21 @@ def __iter__(self):
         for uri, component in self.results_:
             yield uri, component
 
+    def __add__(self, other):
+        cls = self.__class__
+        result = cls()
```

Comment:
Can we find a way to make sure `result` is initialized with the same options (e.g. `collar` and `skip_overlap` for `DiarizationErrorRate` instances) as `self` and `other`? 

This probably means adding some kind of `sklearn`-like [mechanism to clone](https://scikit-learn.org/stable/modules/generated/sklearn.base.clone.html#sklearn.base.clone) metrics.

Reply:
Good point! I cribbed this PR from some monkey patching I did a couple of years ago for an internal SAD scoring tool and in that context, default parameter values were being used, so the issue didn't come up. After looking at how `sklearn` handles this, maybe we could add a similar method to ensure the resulting instance is initialized with the same arguments as the first summand. If so, I should also document that `sum([m1, m2, ...])` assumes all metrics were initialized identically (reasonable).

Reply:
I've made an initial attempt at a `clone` function, which also required me to implement a `get_params` method for metrics. **NOTE** that `get_params` makes the assumption that all subclasses of `BaseMetric` include `**kwargs` in their signature and pass these keyword arguments to the constructor of the superclass (or if multiple superclasses, to one of them). Should this assumption be violated, weirdness could ensue.

An alternate approach would be that used within `sklearn`, which bans use of `*args` and `**kwargs` within constructors and forces each metric to be explicit about its parameters. This would require touching more lines of the codebase, but  beyond being a bit of a chore, shouldn't be difficult to implement.